### ISSUE TITLE
API Proposal: Rename attributes to labelClusters and add in a labels endpoint

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#3290]
I have two related proposals.

First, I suggest that we rename 'access attributes' to 'label clusters.' I think the latter is more clear and self-defining. So, we would shift `/v2/access/attributes` to `/v2/access/labelClusters`. In fact, for background compatibility, perhaps we make this `v3`—in other words,`/v3/access/labelClusters` (so that the `/v2/` stuff stays the same but I leave this up to @misaugstad). I think including a flag to include the raw labels is still appropriate. So, for example, `/v3/access/labelClusters?includeRawLabels=true`

Second, I suggest adding in a new API that is simply `/v3/access/labels`. Yes, we have `attributesWithLabels` (or, as I suggest above, `labelClusters?includeRawLabels=true`) but it's not so easy to iterate through the raw labels. Moreover, perhaps one wants to use a different clustering algorithm. In this case, they'd prefer just getting the raw labels.

I think we still encourage peopled to use `/v3/access/labelClusters` as the default.

**Comment by jonfroehlich** [comment]
I guess I might also think it's more straightforward to have the url be `/v3/api/` than `/v3/access`

**Comment by misaugstad** [comment]
>I guess I might also think it's more straightforward to have the url be `/v3/api/` than `/v3/access`

Agreed. Was going to comment something similar after reading the initial post. Not sure the word "access" is doing anything for us.

>First, I suggest that we rename 'access attributes' to 'label clusters.'

I like it!

>I suggest adding in a new API that is simply `/v3/access/labels`

Agreed. We should also include a few filtering flags. And we should probably have the defaults for the filters to be set up so that you would get the same set of labels that you would get from `/v3/api/labelClusters?includeRawLabels=true`, would you agree with that @jonfroehlich? Here are the filters:
1. tutorial labels
2. labels from "low quality" users that haven't been marked as "correct" by validations
3. labels from "high quality" users that have been marked "incorrect" by validations

**Comment by jonfroehlich** [comment]
Nice, yes. I agree with you to have same default filters.

**Comment by jonfroehlich** [comment]
Also, then we should change `Attribute ID` to `Cluster ID`:
![image](https://github.com/ProjectSidewalk/SidewalkWebpage/assets/1621749/997fb3aa-0976-4d8e-81f7-e1af3b75b83c)

**Comment by jonfroehlich** [comment]
And there are other relevant things that should be renamed to cluster:
![image](https://github.com/ProjectSidewalk/SidewalkWebpage/assets/1621749/ab9a232c-36af-4042-a7b5-b81f3ac26884)

**Comment by jonfroehlich** [comment]
This is quite relevant for our discussions with the Zurich team right now and I still think this is a better naming scheme.

**Comment by misaugstad** [comment]
>In fact, for background compatibility, perhaps we make this v3—in other words,/v3/access/labelClusters (so that the /v2/ stuff stays the same but I leave this up to @misaugstad)

My initial thoughts on this are that we can keep it as v2, but we keep the old API names pointing to the same thing. So `/v2/access/labelClusters` and `/v2/access/attributes` will both work and give the same result. Our switch from v1 to v2 involved a big change in our underlying clustering algorithm, whereas this would just be a simple renaming.

**Comment by misaugstad** [comment]
An update: we've added a new Raw Labels endpoint. Though the renaming of the clustered ones is still an open TODO.

**Comment by misaugstad** [comment]
Closing via #3894 

The endpoint has been renamed! We still have some remaining to do on the back end, but that's covered by issue #3863.
